### DISCUSSION TITLE
no more access to the folders //android/data/net.osmas.plus after upgrade to Android 14 (and security patch)

### DISCUSSION THREAD

**Discussion by PPilotPirx** [osmandapp/OsmAnd#19094] [Q&A]
After upgrading my Sony Xperia 5 IV, I no longer have access to the folders mentioned above. 
Previously, I was able to access the folders with a file manager after approval and, for example, download the photos, videos or voice memos and continue to use them on my PC. 
This is now only possible in a very cumbersome way. For example by viewing a photo with the "files by Google" program, then sharing the photo with this program to another software and saving it somewhere else. Or after closing the image in "files by Google" I have access to a dialogue to save the image somewhere else. 
Would it be possible to build a function into OSMAND to save the self-generated data directly somewhere else?

BTW: The dialog in preferences - data folder - manual specified does not work
System:  Sony XQ-CQ54 
Android version: 14
Build-Number: 64.2.A.2.112
OSMAND: 4.7.1

Thanks PPilot Pirx

**Comment by mariush444** [comment]
It depends what you want to achieve.
You can export from osmand every things using osmand settings -> export to file.
You can import gpx (favorities and tracks) via appropriate menu.
You can import some files via osf (osf is ziped file)

But the first is to use
Android/**media**/net.osmand.plus/files
instead data
the second is to use mixplorer
the best than any othet
...
but the general problem is Android. Newer then worst. I'm still with 11 and I hope my battery will be good for next years :)

**Reply by PPilotPirx** [reply]
In the "export to file" I didn't found an option to export photos you have taken yourself.

**Comment by mwerle** [comment]
Annoying as all <bleep>.

To work around it, luckily OsmAnd allows you to change the data folder to one which you -do- have full access to.

"Menu" -> "Settings" -> "OsmAnd Settings" -> "Data Storage Folder" -> "Manually Specified" and then type in a folder which you have unrestricted access to.   I chose "Documents/GPS/OsmAnd"

OsmAnd will move all of your existing files to the new folder (this will take some time). After that, you can access all files.

**Reply by PPilotPirx** [reply]
This is exactly what does not work with the version of Android I specified above. I can't access the default folders either, I can't change the "Manually Specified" path in the dialogue.

**Reply by mwerle** [reply]
Hmm.. I was able to do it.. and the only reason I looked for the setting and did it was because I could no longer access the files under "/Android/data/net.osmand"..

**Reply by mariush444** [reply]
You can't use Android/data... because Android restriction.

Use Android/media/net.osmand.plus/
in phone memory or sdcard

**Comment by PPilotPirx** [comment]
> You can't use Android/data... because Android restriction.
> 
> Use Android/media/net.osmand.plus/ in phone memory or sdcard

All data under //android/... are locked, access is (no longer) possible. Not even, as before, with explicit release via the Android file system.  This applies to both the smartphone memory and the card memory.

**Reply by mariush444** [reply]
!?
I tested andro14 on galaxy A24. It is accessible. Just use other file manager e.g. total commander or mixplorer.

**Reply by PPilotPirx** [reply]
I have been using TotalCommander on my smartphones for a long time. I don't know Mixexplorer. Totalcommander could no longer access certain memory areas after an update/upgrade years ago. It then offered to allow folder access via a system dialogue. This no longer works. The selection button for sharing is greyed out.
I also tried again to move the data folder to android/media. OSMAnd does something, reports: xxx MB data moved, but when I check with Totalcommander folder is empty.

**Reply by DmitryAlexei** [reply]
@PPilotPirx sometimes you need to change data folder for the second time to move your files to a new destination.

**Comment by PPilotPirx** [comment]
I bought mixplorer but there is no change in behavior. Also mixexplorer does not get permission to access to /android/data.

**Reply by mwerle** [reply]
move the data out of /android..

**Comment by PPilotPirx** [comment]
Once again: it doesn't work. I create a folder "Osmand" or "net.osmand+" under /storage/emulated/o, enter the path in the dialogue in the OSMAND settings, press "Move", a window appears with the move operation and finally the message appears: Maps could not be created in the directory.
Tried several times with different paths. It does not work. 
Disclaimer:  I use the German settings. It may be that various words displayed by OSMAND or Android are called differently in the English-language version.

**Reply by mariush444** [reply]
hmmm...
Probably there are 2 problems
1. Storage path
2. Moving thata by osmand

add 1. I use android/media ( not android/data )
![Screenshot_2024-02-20-14-46-18-47_014aeb6b5e04ee7837dcdaecc5244511](https://github.com/osmandapp/OsmAnd/assets/66887280/f2a18112-1a8c-407b-aa68-073adea0fc43)

It is accessible by total commander and osmand and mixplorer on my phone
Android14

add 2. Moving data with osmamd it is not good idea. I lost my data few times because of osmamd.
If you need your data just export them.
Then just choose path media without moving.

**Reply by PPilotPirx** [reply]
Please, read my answer from yesterday.

**Comment by tarkan123** [comment]
Using  Android/media/  comes with a performance penalty. 

When I moved my osmand folder to under Android/media ,  the GPX tracks and favorites took very long to appear. So I moved back.

Samsung phones allow access to the Android/data when connected via USB, and that is how I copy my Osmand folder for backup purposes.

**Reply by mariush444** [reply]
Android/media can be stored on SD or phone. SD can be slower but not phone.

**Reply by tarkan123** [reply]
No this is on internal storage (I do not have an SD card in the phone).

I think this is a known issue, that file access to the Android/media has a performance hit.

**Comment by XANi** [comment]
Same problem, I can't get the recorded routes out of osmand dir, and I can't point to any other dir that is accessible to file manager

**Reply by scaidermern** [reply]
Can you explain again why using .../Android/media/... is not an option?

**Reply by XANi** [reply]
When I try I get "maps could not be created in specific directory", dialogue says "100% complete, a restart is required" (and it does that "migration" instantly), and when I click the restart button it restarts, and storage is back to "Multiuser Storage 1" and the dir I chose to migrate to is empty
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/c6cdd3dc-3410-4d5e-81c7-79e8b5e04091"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/14c79680-bc87-45d6-a6db-5f699fe5c996"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/6b9b9cbb-37c7-4a8d-9194-ee6c03545938"  width="300" height="auto"/> </td>
  </tr>
</table>

**Reply by scaidermern** [reply]
Try to use the same path as listed for "External Storage 1" but replace "data" with "media. In your case the path should be `/storage/emulated/0/Android/media/net.osmand/files/`. You tried to use `.../osmand/` instead of `.../net.osmand/files/`, maybe this is the reason why it didn't work. Just a guess.

**Reply by XANi** [reply]
That actually worked (to be exact I set `/storage/emulated/0/Android/media/net.osmand/`), and I can access it from file manager, thanks!

Why android have to be so fucking weird... I also tried dirs from outside Android for similar no-effect.

**Reply by scaidermern** [reply]
Great :)

Yes, Android is becoming more and more unusuable :(

**Reply by XANi** [reply]
I think there is some error being eaten here by UI, as it should alert if moving to dir with no access failed

**Reply by scaidermern** [reply]
True. Moving files should result in an error.
And it would be great if OsmAnd could just offer the media directory as storage option.

**Comment by norrsken03** [comment]
OsmAnd 5.1.9 @ Android 14 @ Samsung Galaxy A52 5G

The method according to **mariush444** on Feb 20, 2024, works. The directory name .../Android/media/net.osmand.plus must be strictly adhered to. A directory with a different name or in a different location cannot be created.
However, it is still possible to create an accessible directory with any name and location. There is a bug in OsmAnd that causes file access not to be set automatically after installing the program and, unfortunately, sometimes after updating it. It must be set manually:
Phone | Settings | Apps | three vertical dots at the top | Special access | All files access | OsmAnd | turn on
Then you can create and name a directory anywhere in the file manager and set its path in OsmAnd. Example: /storage/emulated/0/MyOsmAnd
